src/node_http2.cc

```diff
@@ -1439,25 +1424,17 @@ void Http2Session::HandleOriginFrame(const nghttp2_frame* frame) {
   nghttp2_extension ext = frame->ext;
   nghttp2_ext_origin* origin = static_cast<nghttp2_ext_origin*>(ext.payload);
 
-  Local<Value> holder = Array::New(isolate);
-  Local<Function> fn = env()->push_values_to_array_function();
-  Local<Value> argv[NODE_PUSH_VAL_TO_ARRAY_MAX];
+  size_t nov = origin->nov;
+  std::vector<Local<Value>> origin_v(nov);
 
-  size_t n = 0;
-  while (n < origin->nov) {
-    size_t j = 0;
-    while (n < origin->nov && j < arraysize(argv)) {
-      auto entry = origin->ov[n++];
-      argv[j++] =
-        String::NewFromOneByte(isolate,
-                               entry.origin,
-                               v8::NewStringType::kNormal,
-                               entry.origin_len).ToLocalChecked();
-    }
-    if (j > 0)
-      fn->Call(context, holder, j, argv).ToLocalChecked();
+  for (size_t i = 0; i < nov; ++i) {
+    auto entry = origin->ov[i];
```

Comment:
I know this is copy-pasted, but I think getting rid of this `auto` might make things more readable :)